packages/node_modules/pouchdb-adapter-http/src/index.js

```diff
@@ -680,65 +698,69 @@ function HttpPouch(opts, callback) {
       blob = binary ? binStringToBluffer(binary, type) : '';
     }
 
-    // Add the attachment
-    fetchJSON(url, {
-      headers: new Headers({'Content-Type': type}),
-      method: 'PUT',
-      body: blob
-    }, callback).catch(callback);
+    try {
+      // Add the attachment
+      const result = await fetchJSON(url, {
+        headers: new Headers({'Content-Type': type}),
+        method: 'PUT',
+        body: blob
+      });
+      callback(null, result.data);
+    } catch (error) {
+      callback(error);
+    }
   });
 
   // Update/create multiple documents given by req in the database
   // given by host.
-  api._bulkDocs = function (req, opts, callback) {
+  api._bulkDocs = adapterFun('_bulkDocs', async function (req, opts, callback) {
```

Comment:
What's the reason that this method has been wrapped in `adapterFun()` but others like `_put()` have not?

Reply:
We found two problems here.
First, `adapterFun()` should be used on public methods only, so we removed it.
Second, when we attempted to refactor this into async-await, it failed. James pointed out it was [this `return` keyword](https://github.com/pouchdb/pouchdb/pull/8489/commits/3fb18acf9b5d12c888702f7e660890db448cea51#diff-304890c65c36e613ccffc19b7ae5376489e99355d5c84fb318e4ac2c41c75e2aL833), which shouldn’t be needed anymore as all the `_bulkDocs()` implementations in various adapters return nothing, and [this call](https://github.com/pouchdb/pouchdb/blob/701f96dc7c4ba2a598f46381b645a5c4c399b154/packages/node_modules/pouchdb-core/src/adapter.js#L833-L851) is the last statement in the function where it appears. 

This error was caused because the refactor makes `_bulkDocs` to return a promise. The [adapterFun](https://github.com/pouchdb/pouchdb/blob/2eca392bfae8c76bfb8d25b75414e5b47381a653/packages/node_modules/pouchdb-core/src/adapter.js#L769) calls the wrapped function [here](https://github.com/pouchdb/pouchdb/blob/2eca392bfae8c76bfb8d25b75414e5b47381a653/packages/node_modules/pouchdb-utils/src/adapterFun.js#L47) and then that result is wrapped by `toPromise()`. `toPromise()` calls the function and checks if the result is a promise [here](https://github.com/pouchdb/pouchdb/blob/2eca392bfae8c76bfb8d25b75414e5b47381a653/packages/node_modules/pouchdb-utils/src/toPromise.js#L26-L29), and if so, calls the wrapping promise's `fulfill()`, which causes the callback's `fulfill()` to be ignored.